v3/integration/lints/lint/lint.go

```diff
@@ -0,0 +1,139 @@
+package lint
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+type Lint interface {
+	Lint(tree *ast.File, file *File) *Result
+	CheckApplies(tree *ast.File, file *File) bool
+}
```

Comment:
I appreciate that you made this nice and generic :-) Almost like a little mini ZLint inside of ZLint :laughing:

Reply:
Why fix what ain't broke! I saw iit n ZLint and I saw it in that other Go linter that we have integrations with, so it seems like a pretty solid way to accomplish the idea in Go.